### PR TITLE
config(tidb): enable blockade for tidb repo

### PR DESCRIPTION
**Pull request by Mini256**
According to the discussion results in：https://internals.tidb.io/t/topic/374/2

### CODE REVIEW

File: prow/config/plugins.yaml
Comment (tisonkun) [inline prow/config/plugins.yaml:130]:
```suggestion
      This PR modifies significant config files which require review from domain experts from anyone of
      
      @dongyuusa, @bb7133, @XuHuaiyu, @cfzjywxk, or @morgo
      
      You can request review from anyone of them by `/cc @reviewer`
```

cc @zhangyangyu please also take a look on the expression.

File: prow/config/external_plugins_config.yaml
Comment (tisonkun) [inline prow/config/external_plugins_config.yaml:1134]:
```suggestion
        message: Illegal modification of do-not-merge/blocked-paths label.
```

File: prow/config/plugins.yaml
Comment (tisonkun) [inline prow/config/plugins.yaml:129]: I have a question on this list as commented in the discussion thread[1], please do not merge yet.

[1] https://internals.tidb.io/t/topic/374/9

Comment (Mini256) [reply prow/config/plugins.yaml:129]: 👌

Comment (tisonkun) [reply prow/config/plugins.yaml:129]: @Mini256 according to the discussion, @qw4990 is proposed to switch @dongyuusa .

Please also update your work on the thread so that the audiences can know you're doing the work.

File: prow/config/plugins.yaml
Comment (zhangyangyu) [inline prow/config/plugins.yaml:131]: Since we already have a concrete list of domain experts to ask review for, maybe we can don't bother the author to manually type, just auto add the experts to review list?

Comment (Mini256) [reply prow/config/plugins.yaml:131]: Maybe the current configuration is feasible because GitHub will send the mentioned notification to all domain experts when the PR is opened (because they are @) to remind them to pay attention to this PR. 

Sending a requested review in advance will cause unnecessary pending review.

When the other merge conditions of the PR are satisfied, the domain experts still have no response. The contributor or the committer can manually `/cc` request the review of the domain experts.

Comment (zhangyangyu) [reply prow/config/plugins.yaml:131]: I mean remove the @ part and directly add these experts to review list. So no double notification and maybe this entire explain message is not needed. Current message gives me a feeling that I should `\cc` immediately after the message is prompted and then double notification.

Comment (Mini256) [reply prow/config/plugins.yaml:131]: @tisonkun, what do you think?

The blockade plugin doesn't support the auto-request review. Auto request review is usually the responsibility of the [blunderbuss](https://book.prow.tidb.io/#/plugins/blunderbuss) plugin. If it needs to be automatically allocated according to the file path, need to introduce the OWNERS file mechanism.

for example:
https://github.com/kubernetes/test-infra/blob/03e3720d1199cbd13462b82a5f545c544799ccd8/config/prow/OWNERS#L3

Comment (tisonkun) [reply prow/config/plugins.yaml:131]: @Mini256 I don't have a strong feeling on request review or not. But stuck at so-called upstream limitation is undesirable for tichi to handle the PR review process.

We cannot say because of k8s fault, we stuck. If so, think of a resolution to avoid such unreliable dependencies.

File: prow/config/external_plugins_config.yaml
Comment (tisonkun) [inline prow/config/external_plugins_config.yaml:1133]: is it possible a random user remove the do-not-merge label by `/unlabel do-not-merge/blocked-paths`?

Comment (Mini256) [reply prow/config/external_plugins_config.yaml:1133]: No, because do-not-merge type labels are not in the trusted list, the robot will not respond.

https://github.com/ti-community-infra/configs/blob/b7c9df395a9819df91660364cdf0f518e5086e3c/prow/config/external_plugins_config.yaml#L423

File: prow/config/plugins.yaml
Comment (Mini256) [reply prow/config/plugins.yaml]: @tisonkun, the blockade plugin can request all the domain experts by `/cc` command, but cannot select one of them to request.

@zhangyangyu, updated in the new commit.

File: prow/config/plugins.yaml
Comment (tisonkun) [inline prow/config/plugins.yaml:129]: @qw4990 @bb7133 @XuHuaiyu @cfzjywxk @morgo Please be aware of this change. If a PR touches config files, _all_ of you will receive a review request.

**Review by zhangyangyu — APPROVED** [review]

**Pull request merged**
